src/editors/containers/TextEditor/components/SelectImageModal/index.jsx

```diff
@@ -45,36 +57,56 @@ export const SelectImageModal = ({
       )}
       title={intl.formatMessage(messages.titleLabel)}
     >
-
       {/* Error Alerts */}
       <FetchErrorAlert />
       <UploadErrorAlert />
+      <ErrorAlert
+        dismissError={inputError.dismiss}
+        hideHeading
+        isError={inputError.show}
+      >
+        <FormattedMessage {...messages.fileSizeError} />
+      </ErrorAlert>
 
       {/* User Feedback Alerts */}
       <ErrorAlert
-        dismissError={error.dismiss}
+        dismissError={galleryError.dismiss}
         hideHeading
-        isError={error.show}
+        isError={galleryError.show}
       >
         <FormattedMessage {...messages.selectImageError} />
       </ErrorAlert>
-
       <Stack gap={3}>
         <SearchSort {...searchSortProps} />
-        <Gallery {...galleryProps} />
+        {!inputIsLoading ? <Gallery {...galleryProps} /> : (
+          <Spinner
+            animation="border"
+            className="mie-3"
+            screenReaderText={intl.formatMessage(messages.loading)}
+          />
+        )}
         <FileInput fileInput={fileInput} />
       </Stack>
     </BaseModal>
-
   );
 };
 
 SelectImageModal.propTypes = {
   isOpen: PropTypes.bool.isRequired,
   close: PropTypes.func.isRequired,
   setSelection: PropTypes.func.isRequired,
+  clearSelection: PropTypes.func.isRequired,
   // injected
   intl: intlShape.isRequired,
+  // redux
+  inputIsLoading: PropTypes.bool.isRequired,
 };
 
-export default injectIntl(SelectImageModal);
+const requestKey = RequestKeys.uploadImage;
+export const mapStateToProps = (state) => ({
+  inputIsLoading: selectors.requests.isPending(state, { requestKey }),
```

Comment:
```suggestion
  inputIsLoading: selectors.requests.isPending(state, { RequestKeys.uploadImage }),
```
and delete line 105.

Reply:
Note that for similar use cases, I have used `selectors.requests.isFinished(state, { requestKey: RequestKeys.fetchBlock })`, instead of isPending, but both semantically do the same thing. You might imagine some marginal benefit from that being consistent.